libraries/AP_HAL/DSP.cpp

```diff
@@ -247,7 +285,7 @@ float DSP::calculate_jains_estimator(const FFTWindowState* fft, const float* rea
 bool DSP::fft_init_average(FFTWindowState* fft)
 {
     if (fft->_avg_freq_bins == nullptr) {
-        fft->_avg_freq_bins = (float*)hal.util->malloc_type(sizeof(float) * (fft->_window_size), DSP_MEM_REGION);
+        fft->_avg_freq_bins = (float*)hal.util->malloc_type(sizeof(float) * (fft->_bin_count + 1), DSP_MEM_REGION);
```

Comment:
some explanations of the +1 would be good

Reply:
Fixed